_posts/2019-12-05-elasticsearch-percolator-qualification.md

```diff
@@ -0,0 +1,198 @@
+---
+layout: post
+title:  "Elasticsearch percolator use case for document classification"
+excerpt: ""
+date:   2019-12-05 12:00:00 +0100
+categories: python
+tags: python elasticsearch percolator
+---
+
+Currently at Allo-Media, we are using Elasticsearch in its general workflow which is to create an index and store documents holding our calls metadata, and then allowing to search through these documents given some business criteria like : "Give me all calls from client Acme, where the customer speaks about the french strike".
+
+The percolator feature from Elasticsearch allows to make a reverse search. We store search queries as documents in its own index, and then we can percolate new call documents and retrieve what search queries match. One use case to use the percolator is document classification.
+
+For example, say that we want to tag with `Check sent` all calls mentionning that the user has already sent a bank check. We would have the following search query : 
+```
+("I've sent" | "I've already sent") ("check")
```

Comment:
So we actually search in transcripts, so instead of *phone call metadata* you may just want to use *audio transcripts*.